`ReflectionUtil.findNonSyntheticMethod` failing for non-generics methods

## Whats your runtime?
* Dozer version: 6.1.0
* OS version: CentOS 6.5 / Mac OS High Sierra
* JDK version: 1.8u131
  
## Whats the problem?
`ReflectionUtil.findNonSyntheticMethod` is failing (resulting in `null` return value) in valid scenario.

### Observed Results:
[PageImpl](https://github.com/spring-projects/spring-data-commons/blob/1.13.x/src/main/java/org/springframework/data/domain/PageImpl.java) is a Spring Data class representing a paged collection from DB. It extends [Chunk](https://github.com/spring-projects/spring-data-commons/blob/1.13.x/src/main/java/org/springframework/data/domain/Chunk.java) which in turn implements [Slice](https://github.com/spring-projects/spring-data-commons/blob/1.13.x/src/main/java/org/springframework/data/domain/Slice.java).

I try to map [`Page<MyEntity>`](https://github.com/spring-projects/spring-data-commons/blob/1.13.x/src/main/java/org/springframework/data/domain/Page.java) to a DTO and get this error during mapping:
```
Caused by: java.lang.NoSuchMethodException: Unable to determine read method for Field: 'number' in Class: class org.springframework.data.domain.PageImpl
	at org.dozer.propertydescriptor.JavaBeanPropertyDescriptor.retryMissingMethod(JavaBeanPropertyDescriptor.java:94) ~[dozer-core-6.1.0.jar:6.1.0]
	at org.dozer.propertydescriptor.JavaBeanPropertyDescriptor.getReadMethod(JavaBeanPropertyDescriptor.java:68) ~[dozer-core-6.1.0.jar:6.1.0]
	at org.dozer.propertydescriptor.JavaBeanPropertyDescriptor.retryMissingMethod(JavaBeanPropertyDescriptor.java:99) ~[dozer-core-6.1.0.jar:6.1.0]
	at org.dozer.propertydescriptor.JavaBeanPropertyDescriptor.getReadMethod(JavaBeanPropertyDescriptor.java:68) ~[dozer-core-6.1.0.jar:6.1.0]
	at org.dozer.propertydescriptor.GetterSetterPropertyDescriptor.invokeReadMethod(GetterSetterPropertyDescriptor.java:284) ~[dozer-core-6.1.0.jar:6.1.0]
```
After extensive debugging I found that because `!method.isBridge()` and `!method.isSynthetic()` are returning `false` in the `if` condition, the `ReflectionUtil.findNonSyntheticMethod` method returns `null` for get method.

I found this part was introduced in the following PR: https://github.com/DozerMapper/dozer/pull/4/
  
### Expected Results:
I should get the `Method` for `getNumber` and `setNumber` and, in the end, a correct `PropertyDescriptor` in [ReflectionUtils.java#L117](https://github.com/DozerMapper/dozer/blob/master/core/src/main/java/org/dozer/util/ReflectionUtils.java#L117)

## Link to GitHub repo with Unit test
https://github.com/mateusz-bajorek/dozer-test
You can run the test with `./mvnw test`. It will fail with the exception above.

--- a/core/src/main/java/com/github/dozermapper/core/util/ReflectionUtils.java
+++ b/core/src/main/java/com/github/dozermapper/core/util/ReflectionUtils.java
@@ -109,8 +109,8 @@ public final class ReflectionUtils {
             String baseName = Character.toUpperCase(propertyName.charAt(0)) + propertyName.substring(1);
             String setMethodName = "set" + baseName;
             String getMethodName = "get" + baseName;
-            Method getMethod = findNonSyntheticMethod(getMethodName, clazz);
-            Method setMethod = findNonSyntheticMethod(setMethodName, clazz);
+            Method getMethod = findPreferablyNonSyntheticMethod(getMethodName, clazz);
+            Method setMethod = findPreferablyNonSyntheticMethod(setMethodName, clazz);
             try {
                 return new PropertyDescriptor(propertyName, getMethod, setMethod);
             } catch (IntrospectionException e) {
@@ -120,14 +120,19 @@ public final class ReflectionUtils {
         return descriptor;
     }
 
-    private static Method findNonSyntheticMethod(String methodName, Class<?> clazz) {
+    private static Method findPreferablyNonSyntheticMethod(String methodName, Class<?> clazz) {
         Method[] methods = clazz.getMethods();
+        Method syntheticMethod = null;
         for (Method method : methods) {
-            if (method.getName().equals(methodName) && !method.isBridge() && !method.isSynthetic()) {
-                return method;
+            if (method.getName().equals(methodName)) {
+                if (!method.isBridge() && !method.isSynthetic()) {
+                    return method;
+                } else {
+                    syntheticMethod = method;
+                }
             }
         }
-        return null;
+        return syntheticMethod;
     }
 
     public static DeepHierarchyElement[] getDeepFieldHierarchy(Class<?> parentClass, String field,

--- a/core/src/test/java/com/github/dozermapper/core/util/ReflectionUtilsTest.java
+++ b/core/src/test/java/com/github/dozermapper/core/util/ReflectionUtilsTest.java
@@ -197,6 +197,12 @@ public class ReflectionUtilsTest extends AbstractDozerTest {
         assertEquals("java.util.Map", clazz.getCanonicalName());
     }
 
+    @Test
+    public void shouldDetermineReadMethodForSyntheticOnlyMethod() throws Exception {
+        PropertyDescriptor propertyDescriptor = ReflectionUtils.findPropertyDescriptor(ListHolderWrapperImpl.class, "list", null);
+        assertNotNull(propertyDescriptor.getReadMethod());
+    }
+
     public class Y implements HasX<ClassInheritsClassX> {
         private ClassInheritsClassX x;
 
@@ -254,6 +260,33 @@ public class ReflectionUtilsTest extends AbstractDozerTest {
 
     private interface TestIF2 {
         Integer getB();
+    }
+
+    public class ListHolderWrapperImpl<T> extends AbstractListHolder<T> implements ListHolderWrapper<T> {
+        public ListHolderWrapperImpl(List<T> content) {
+            super(content);
+        }
+    }
+
+    abstract class AbstractListHolder<T> implements ListHolder<T> {
+        private List<T> list;
+
+        AbstractListHolder(List<T> list) {
+            this.list = list;
+        }
+
+        @Override
+        public List<T> getList() {
+            return list;
+        }
+    }
+
+    interface ListHolder<T> {
+        List<T> getList();
+    }
+
+    interface ListHolderWrapper<T> extends ListHolder<T> {
+
     }
 
 }

